src/deploy/functions/runtimes/discovery/v1alpha1.ts

```diff
@@ -248,19 +265,21 @@ function assertManifestEndpoint(ep: ManifestEndpoint, id: string): void {
 
 function parseEndpointForBuild(
   id: string,
-  ep: ManifestEndpoint,
+  ep: WireEndpoint,
   project: string,
   defaultRegion: string,
   runtime: runtimes.Runtime
 ): build.Endpoint {
   let triggered: build.Triggered;
-  if (backend.isEventTriggered(ep)) {
+  if (build.isEventTriggered(ep)) {
     const eventTrigger: build.EventTrigger = {
       eventType: ep.eventTrigger.eventType,
       retry: ep.eventTrigger.retry,
     };
     // Allow serviceAccountEmail but prefer serviceAccount
-    renameIfPresent(eventTrigger, ep.eventTrigger, "serviceAccount", "serviceAccountEmail");
+    if ("serviceAccountEmail" in (ep.eventTrigger as any)) {
+      eventTrigger.serviceAccount = (ep.eventTrigger as any).serviceAccountEmail;
+    }
```

Comment:
renameIfPresent is _too_ type-safe, i think--typescript refuses to compile i because build.EventTigger doesn't _have_ a serviceAccountEmail field; we're trying to capture a legacy field which isn't part of our currently defined wire format

Reply:
You _could_ just define it in the wire format with a `// deprecated` comment.